build.gradle

```diff
@@ -21,6 +21,10 @@ idea.project.settings {
 
 import net.ltgt.gradle.errorprone.CheckSeverity
 
+project('jib-maven-plugin-extension-api') {
+  apply plugin: 'java-library'
```

Comment:
Should I apply `java-library` to use the `api` dependencies? I didn't on `jib-build-plan`. This kind of stuff is always confusing now that we need to publish JARs from `jib-build-plan` and this sub-project.

Reply:
I mean if `java` lets you define `api` dependencies, then maybe this is unnecessary?

Reply:
`java` doesn't define `api`. `java-library` is necessary.

Reply:
yeah I dunno if there's a better way to do this, I don't like applying these plugins conditionally in some parent. But maybe this is okay for now.

Reply:
Oh, maybe I should put this in `jib-maven-plugin-extension-api` instead at root.

Reply:
Nah, there was a reason I put it at root. Looks like `java-library` needs to be applied before `java`.

Reply:
oh yeah I agree, I think maybe there's a consequence to us putting so much in the root buildfile :/